### PR TITLE
ui v2: text field

### PR DESCRIPTION
**Pull request by danielhochman**
### Description
Implement new text field design with the exception of the "clear" control, which requires some more though and potentially refactoring to standardize on controlled vs uncontrolled components.

### Testing Performed
See stories.

### CODE REVIEW

File: frontend/packages/core/src/Input/text-field.tsx [hunk 1]
```diff
@@ -1,37 +1,122 @@
...
-import type { InputLabelProps, TextFieldProps as MuiTextFieldProps } from "@material-ui/core";
-import { TextField as MuiTextField } from "@material-ui/core";
-import styled from "styled-components";
+import * as React from "react";
+import styled from "@emotion/styled";
+import {
+  InputProps as MuiInputProps,
+  StandardTextFieldProps as MuiStandardTextFieldProps,
+  TextField as MuiTextField,
+} from "@material-ui/core";
```
Comment (dschaller) [inline frontend/packages/core/src/Input/text-field.tsx:7]:
```suggestion
import type {
  InputProps as MuiInputProps,
  StandardTextFieldProps as MuiStandardTextFieldProps,
} from "@material-ui/core";
import { TextField as MuiTextField } from "@material-ui/core";
```

File: frontend/packages/core/src/Input/text-field.tsx [hunk 2]
```diff
@@ -1,37 +1,122 @@
...
+  },
+
+  ".MuiInputBase-root.Mui-error": {
+    borderColor: "#db3615",
+  },
+
+  ".MuiFormHelperText-root > svg": {
+    height: "16px",
+    width: "16px",
+    marginRight: "5px",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/text-field.tsx:84]: It looks like this is 4px in the designs

File: frontend/packages/core/src/Input/text-field.tsx [hunk 3]
```diff
@@ -1,37 +1,122 @@
...
+
+  ".MuiInputLabel-root.Mui-error": {
+    color: "#db3615",
+  },
+
+  ".MuiInputBase-root": {
+    border: "1px solid rgba(13, 16, 48, 0.38)",
+    borderRadius: "4px",
+    fontSize: "16px",
+    padding: "14px 16px",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/text-field.tsx:45]: I believe this padding is 13px on top and bottom in the designs

File: frontend/packages/core/src/Input/text-field.tsx [hunk 4]
```diff
@@ -1,37 +1,122 @@
...
+  },
+
+  "label + .MuiInput-formControl": {
+    marginTop: "20px",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/text-field.tsx:49]: I think this is 22px in the designs

File: frontend/packages/core/src/Input/text-field.tsx [hunk 5]
```diff
@@ -1,37 +1,122 @@
...
+    InputLabelProps={{ ...InputLabelProps, shrink: true }}
+    InputProps={{ ...InputProps, disableUnderline: true }}
+    fullWidth
+    {...props}
+  />
+);
+
+const StyledTextField = styled(BaseTextField)({
+  ".MuiInputLabel-root": {
+    fontSize: "13px",
```
[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:24]: it's roboto medium in figma... so this is the effective match i think

File: frontend/packages/core/src/Input/text-field.tsx [hunk 6]
```diff
@@ -1,37 +1,122 @@
...
+    padding: "0",
+    height: "20px",
+  },
+
+  ".MuiFormHelperText-root": {
+    verticalAlign: "middle",
+    display: "flex",
+    position: "relative",
+    fontSize: "12px",
+    marginTop: "7px",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/text-field.tsx:70]:
```suggestion
    marginTop: "8px",
```

File: frontend/packages/core/src/Input/text-field.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (dschaller) [reply frontend/packages/core/src/Input/text-field.tsx:24]: I think this is slightly more bold than figma. Changing it to 500 looks like 
Screen Shot 2020-12-01 at 10 59 38 AM vs bold as 
Screen Shot 2020-12-01 at 11 00 10 AM

[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:24]: got it, 13/500 looks like a match to me

File: frontend/packages/core/src/Input/text-field.tsx [hunk 7]
```diff
@@ -1,37 +1,122 @@
...
+    fullWidth
+    {...props}
+  />
+);
+
+const StyledTextField = styled(BaseTextField)({
+  ".MuiInputLabel-root": {
+    fontSize: "13px",
+    fontWeight: "bold",
+    transform: "scale(1)",
```
[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:26]: labels have a .75 scale which i revert with this

File: frontend/packages/core/src/Input/text-field.tsx [hunk 8]
```diff
@@ -1,37 +1,122 @@
...
+      | "error"
+      | "helperText"
+      | "id"
+      | "inputRef"
+      | "label"
+      | "multiline"
+      | "name"
+      | "onChange"
+      | "onFocus"
+      | "onKeyDown"
```
[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:102]: you're using onKeyDown in the resolver so i kept it, wasn't sure of implication of changing it and didn't want to change interfaces beyond style props

File: frontend/packages/core/src/Input/text-field.tsx (hunk 2 above)
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx]: the element in figma has 1px padding seen here, 1+4 = 5

Screenshot from 2020-12-01 12-57-42

File: frontend/packages/core/src/Input/text-field.tsx [hunk 9]
```diff
@@ -1,37 +1,122 @@
...
+      | "required"
+      | "type"
+      | "value"
+    >,
+    Pick<MuiInputProps, "readOnly"> {
   onReturn?: () => void;
 }
 
-const TextField: React.FC<TextFieldProps & MuiTextFieldProps> = ({
+export const TextField = ({
```
[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:120]: storybook needs this export to render full props

File: frontend/packages/core/src/Input/text-field.tsx [hunk 10]
```diff
@@ -43,26 +128,24 @@ const TextField: React.FC<TextFieldProps & MuiTextFieldProps> = ({
...
-
-  const placeholderInputTypes = ["date", "datetime-local", "month", "time", "week"];
-  const hasInputPlaceholder = placeholderInputTypes.indexOf(type) !== -1;
-  if (placeholder || hasInputPlaceholder) {
-    inputLabelProps.shrink = true;
+  // Prepend a '!' icon to helperText displayed below the form if the form is in an error state.
+  if (error) {
+    helperText = (
+      <>
+        <ErrorIcon />
```
[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:143]: fixed

File: frontend/packages/core/src/Input/text-field.tsx (hunk 3 above)
Comment (danielhochman) [reply frontend/packages/core/src/Input/text-field.tsx:46]: figma has line height, for multiline content which is usually dense (e.g. JSON) didn't want line height, this padding is a pixel match with figma

File: frontend/packages/core/src/Input/text-field.tsx [hunk 11]
```diff
@@ -1,37 +1,130 @@
...
+
+  ".MuiInputBase-root.Mui-error": {
+    borderColor: "#db3615",
+  },
+
+  ".MuiFormHelperText-root > svg": {
+    height: "16px",
+    width: "16px",
+    marginRight: "4px",
+    // alignSelf: "baseline",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/text-field.tsx:92]:
```suggestion
```

File: frontend/packages/core/src/Input/text-field.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (dschaller) [reply frontend/packages/core/src/Input/text-field.tsx:110]: 👍  we should figure out a single callback but it can be done in a separate PR

**Review by dschaller — APPROVED** [review]
pending removing commented out code

**Pull request merged**